### ISSUE TITLE
Few questions (training+evaluation)

### ISSUE THREAD

**Issue by caiocj1** [OpenDriveLab/BeTop#5]
Hello,

Thank you for the awesome work! I had just a few questions.

I did a training run (not using --fix_random_seed and using a global batch size of 40 due to hardware limitation) and the joint results on the interactive validation set are not as good as in the paper:
- mAP: 2248 here vs. 0.2366 paper
- minFDE: 2.581 here vs. 2.134 paper

Since the training configuration is not exactly the same a difference is to be expected but it seems too large to me. Just to make sure, I wanted to ask:
1. Did you observe high sensitivity to seed and batch size in your experiments?
2. To use the exact same evaluation you use in the paper, you simply use the interactive validation dataset and set joint_pred to true when making the final predictions, correct?

Thank you in advance!

**Comment by georgeliu233** [comment]
Hi, thanks for your question! 

Batch size appears to be a factor in sensitivity, as it depends on the specific scenario. I also noticed the performance variations when adjusting the batch size during previous MTR training. Therefore, I recommend caching the preprocessed data and choosing an appropriate batch size for training. As the batch_size is limited, I think you could also try to enlarge the number of topo agents during decoding, to compensate for agent info.  

Another potential issue is that the Val results we report comes directly from the WOMD portal. It seems they have had slight modifications to the joint mAP calculation compared to the womd package. To ensure accuracy, I suggest submitting your val results to the portal for verification.

**Comment by caiocj1** [comment]
Thank you for the help, the issue was actually that from what I understood NMS wasn't being applied in the final predictions for evaluation, so I fixed that and most of the performance recovered.

I have another question, I saw that BeTop is using the first version of the MTR encoder, are there any plans to update the repo so it uses the MTR++ version?

**Comment by georgeliu233** [comment]
Hi, thanks for your update. I have tried to reproduce the MTR++ but it does not perform as well as it reported. Currently I'm devoting to another project, so it is appreciated to try on the deprecated MTR++ code, and we would be happy to update this part if reported any improvements using MTR++

Best
